lib/sensors/GPS.cpp

```diff
@@ -0,0 +1,494 @@
+#include "GPS.h"
+GPS::GPS(PinName tx, PinName rx, uint32_t baud_rate) : m_uart(tx, rx, DEFAULT_BAUD_RATE) {
+	m_rollingChecksumA = 0;
+	m_rollingChecksumB = 0;
+	m_ubxFrameCounter = 0;
+	m_ignoreThisPayload = false;
+	m_ack_flag = false;
+	m_nack_flag = false;
+	m_target_class = UBX_CLASS_NAV; //default class target
+	m_target_msgID = UBX_NAV_PVT; //default id target
+	m_buf_write = m_buf_read = m_buffer.begin();
+	setSerialRate(baud_rate, COM_PORT_UART1);
+	m_uart.baud(baud_rate);
+	wait_ms(50); //ACK lost during baud rate change
+	m_uart.attach(callback(this, &GPS::onByteReceive));
+}
+
+void GPS::configure() {
```

Comment:
is this supposed to be like an init?

Reply:
Yes

Reply:
but it doesn't need to get called because it auto loads stuff from flash?

Reply:
It will be necessary to store that data in the internal flash to begin with (only the current config is stored when you call the save function). Also I'm not entirely clear on where the config is actually stored, if its in battery-backed RAM then the save will only be valid for as long as the supercap lasts.

Reply:
which is how long? should we just be always calling configure then?

Reply:
It's around 3 days, but it could be safer to just always call configure.